integration-auth/src/main/java/org/symphonyoss/integration/authentication/jwt/JwtAuthentication.java

```diff
@@ -58,13 +98,56 @@ public String getJwtToken(String authorizationHeader) {
    */
   public Long getUserId(String token) {
     if (StringUtils.isEmpty(token)) {
-      String message = logMessageSource.getMessage(JWT_TOKEN_EMPTY);
-      String solution = logMessageSource.getMessage(JWT_TOKEN_EMPTY_SOLUTION);
+      String message = logMessage.getMessage(JWT_TOKEN_EMPTY);
+      String solution = logMessage.getMessage(JWT_TOKEN_EMPTY_SOLUTION);
       throw new UnauthorizedUserException(message, solution);
     }
 
     // TODO APP-1206 Need to be implemented
     return new Long(0);
   }
 
+  /**
+   * Start the JWT authentication between the App and the SBE.
+   * @param configurationId Application identifier.
+   * @return The generated Application Token (Ta).
+   */
+  public String authenticate(String configurationId) {
+    String appToken = tokenUtils.generateToken();
+    AppToken bothTokens = appAuthenticationService.authenticate(configurationId, appToken);
+
+    Integration integration = integrationBridge.getIntegrationById(configurationId);
+    String sessionToken = authenticationProxy.getSessionToken(integration.getSettings().getType());
+    try {
+      apiClient.saveAppAuthenticationToken(sessionToken, configurationId, bothTokens);
+    } catch (RemoteApiException e) {
+      throw new UnexpectedAuthException(logMessage.getMessage(API_MSG), e,
+          logMessage.getMessage(API_MSG_SOLUTION));
+    }
+
+    return appToken;
+  }
+
+  /**
+   * Get the Symphony previously generated token by the app token
+   * @param configurationId Application identifier.
+   * @param applicationToken App token generated by the "authenticate" service.
+   * @return The generated Token (Ta) and the SBE token (Ts).
+   */
+  public boolean isValidTokenPair(String configurationId, String applicationToken,
+      String symphonyToken) {
+    Integration integration = integrationBridge.getIntegrationById(configurationId);
+    String sessionToken = authenticationProxy.getSessionToken(integration.getSettings().getType());
+    try {
+      AppToken bothTokens = apiClient.getAppAuthenticationToken(sessionToken, configurationId,
+          applicationToken);
+      return symphonyToken.equals(bothTokens.getSymphonyToken());
+    } catch (RemoteApiException e) {
+      if (e.getCode() == HttpStatus.SC_NOT_FOUND) {
+        return false;
+      }
+      throw new UnexpectedAuthException(logMessage.getMessage(API_MSG), e,
```

Comment:
I think you should raise this exception inside the API client rather than raise the RemoteApiException.